app.py

```diff
@@ -52,25 +53,26 @@ def confirm_login():
 def register():
    return render_template("register.html")
 
-
 @app.route("/register/resultsjson", methods = ['POST','GET'])
 def resultsJSON():
    if request.method == 'POST':
       results_data = request.form
       # converting dictionary to json
-      data_json = json.dumps(results_data)
+      data_json = json.dumps([results_data])
       # read to register.json file
       with open("register.json","r") as outfile:
          # creating a json object in the form of key/value pair
          account = json.load(outfile)
-         if("email" in account == data_json.get("email")):
+         shared_items = {k: account[k] for k in account if k in data_json and account[k] == data_json[k]}
+         if shared_items == True:
             return f"username already exists"
          else:
             with open("register.json","a+") as f:
                f.write(data_json +"\n")
```

Comment:
We are also not validating what is being saved; I don't see any point where we clean up the inputted data before we save the data.

Reply:
I am really stuck and I don't know how to approach this problem...kindly show me